src/frontend/src/components/TaskList.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Box, Tab, Tabs, Typography } from '@mui/material';
+import { useState } from 'react';
+import PageBlock from '../layouts/PageBlock';
+import { NERButton } from './NERButton';
+import { useAuth } from '../hooks/auth.hooks';
+import { Add } from '@mui/icons-material';
+import { Auth } from '../utils/types';
+
+interface TaskListProps {
+  defaultClosed?: boolean;
+}
+
+// Page block containing task list view
+const TaskList = ({ defaultClosed }: TaskListProps) => {
+  const auth: Auth = useAuth();
+  const taskListTitle: string = 'Task List';
+  const addTaskButton: React.ReactNode = (
+    <NERButton
+      variant="contained"
+      disabled={auth.user?.role === 'GUEST'}
+      startIcon={<Add />}
+      onClick={() => alert("addTaskButton doesn't work yet. If you want to work on it go to issue #762")}
+    >
+      New Task
+    </NERButton>
+  );
+
+  const [value, setValue] = useState<number>(0);
+
+  const handleTabChange = (event: React.SyntheticEvent, newValue: number): void => {
+    setValue(newValue);
+  };
+
+  // TabPanel stuff copied from https://mui.com/material-ui/react-tabs/ unless we install @mui/lab
+
+  interface TabPanelProps {
+    children?: React.ReactNode;
+    index: number;
+    value: number;
+  }
+
+  const TabPanel = (props: TabPanelProps) => {
+    const { children, value, index, ...other } = props;
+
+    return (
+      <div
+        role="tabpanel"
+        hidden={value !== index}
+        id={`simple-tabpanel-${index}`}
+        aria-labelledby={`simple-tab-${index}`}
+        {...other}
+      >
+        {value === index && (
+          <Box sx={{ p: 3 }}>
+            <Typography>{children}</Typography>
+          </Box>
+        )}
+      </div>
+    );
+  };
```

Comment:
I would say, put this TabPanel it's own file. Eventually this'll be changed to be the content inside the taskslist component which will probably be complex enough to warrant it's own file

Reply:
It shouldn't have children then right? I'll call it TaskListTabContents or something

Reply:
yup, looks good. Will approve later after a lil manual testing